### PR TITLE
fix: [ORI-1804] properly return full error code in error message.

### PR DESCRIPTION
**Pull request by avmturo**
# Submit a pull request

## Description of the pull request
Fix bug with error message format
Add tests

### CODE REVIEW

**Comment by avmturo** [comment]
In the example:
{"success":false,"error":{"type":"validation_error","detail":{"code":"required","message":"This field is required.","field_name":"data.image_url"}}}

We would also want to include field_name along this the message

File: original_sdk/base/exceptions.py [hunk 1]
```diff
@@ -14,8 +10,8 @@ def __init__(self, text: str, status_code: int) -> None:
 
         try:
             parsed_response: Dict = json.loads(text)
-            self.error_code = parsed_response.get("code", "unknown")
-            self.error_message = parsed_response.get("message", "unknown")
+            self.error_code = parsed_response["error"]["detail"]["code"]
```
Comment (don1989) [inline original_sdk/base/exceptions.py:13]: Why don't we raise the original exception we receive when we do the request?

File: original_sdk/base/exceptions.py [hunk 2]
```diff
@@ -13,15 +8,15 @@ def __init__(self, text: str, status_code: int) -> None:
...
-            self.error_message = parsed_response.get("message", "unknown")
+            full_error = json.loads(text)
+            self.error_type = full_error["error"]["type"]
+            self.error_detail = full_error["error"]["detail"]
             self.json_response = True
         except ValueError:
             pass
 
     def __str__(self) -> str:
         if self.json_response:
```
Comment (mgom) [inline original_sdk/base/exceptions.py:19]: can you add tests for this case? with / without json_response

**Review by don1989 — APPROVED** [review]
lgtm

**Pull request merged**